packages/mattermost-redux/src/reducers/entities/users.ts

```diff
@@ -171,50 +183,50 @@ function profiles(state: IDMappedObjects<UserProfile> = {}, action: GenericActio
         if (oldUser) {
             user.terms_of_service_id = oldUser.terms_of_service_id;
             user.terms_of_service_create_at = oldUser.terms_of_service_create_at;
+
+            if (isEqual(user, oldUser)) {
+                return state;
+            }
         }
 
         return {
             ...state,
             [data.id]: user,
         };
     }
-    case UserTypes.RECEIVED_PROFILES_LIST:
-        return Object.assign({}, state, profileListToMap(action.data));
-    case UserTypes.RECEIVED_PROFILES:
-        return Object.assign({}, state, action.data);
+    case UserTypes.RECEIVED_PROFILES_LIST: {
+        const users: UserProfile[] = action.data;
 
-    case UserTypes.LOGIN: { // Used by the mobile app
-        const {user} = action.data;
+        return users.reduce((nextState, user) => {
```

Comment:
It's kind of a weird one since we don't really know if it's `state` or `nextState` at this point since the previous iterations could've returned either the existing state or a new one. This is the best naming convention I've come up with so far when using reduce though since just calling it `state` would cause it to shadow the outer one.

Reply:
I'm good with it 👍